src/main/java/com/microsoft/sqlserver/jdbc/SQLServerNoneEnclaveProvider.java

```diff
@@ -0,0 +1,249 @@
+                        enclaveRequestedCEKs);
+                // Process the third result set.
+                if (connection.isAEv2() && stmt.getMoreResults()) {
+                    try (ResultSet hgsRs = stmt.getResultSet()) {
+                        if (hgsRs.next()) {
+                            noneResponse = new NoneAttestationResponse(hgsRs.getBytes(1));
+                            // This validates and establishes the enclave session if valid
+                            validateAttestationResponse();
+                        } else {
+                            SQLServerException.makeFromDriverError(null, this,
+                                    SQLServerException.getErrString("R_UnableRetrieveParameterMetadata"), "0", false);
+                        }
+                    }
+                }
+            }
+        } catch (SQLException | IOException e) {
+            if (e instanceof SQLServerException) {
+                throw (SQLServerException) e;
+            } else {
+                throw new SQLServerException(SQLServerException.getErrString("R_UnableRetrieveParameterMetadata"), null,
+                        0, e);
+            }
+        }
+        return enclaveRequestedCEKs;
+    }
+}
+
+
+class NoneAttestationParameters extends BaseAttestationRequest {
+
+    // Type 2 is NONE, sent as Little Endian 0x20000000
+    private static final byte[] ENCLAVE_TYPE = new byte[] {0x2, 0x0, 0x0, 0x0};
+    // Nonce length is always 256
+    private static final byte[] NONCE_LENGTH = new byte[] {0x0, 0x1, 0x0, 0x0};
+    private final byte[] nonce = new byte[256];
+
+    NoneAttestationParameters(String attestationUrl) throws SQLServerException, IOException {
+        byte[] attestationUrlBytes = (attestationUrl + '\0').getBytes(UTF_16LE);
+
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        os.write(ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt(attestationUrlBytes.length).array());
+        os.write(attestationUrlBytes);
+        os.write(NONCE_LENGTH);
+        new SecureRandom().nextBytes(nonce);
+        os.write(nonce);
+        enclaveChallenge = os.toByteArray();
+
+        initBcryptECDH();
+    }
+
+    @Override
+    byte[] getBytes() throws IOException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        os.write(ENCLAVE_TYPE);
+        os.write(ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt(enclaveChallenge.length).array());
+        os.write(enclaveChallenge);
+        os.write(ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt(ENCLAVE_LENGTH).array());
+        os.write(ECDH_MAGIC);
+        os.write(x);
+        os.write(y);
+        return os.toByteArray();
+    }
+
+    byte[] getNonce() {
+        return nonce;
+    }
+}
+
+
+@SuppressWarnings("unused")
+class NoneAttestationResponse extends BaseAttestationResponse {
+
+    private static final Hashtable<String, JWTCertificateEntry> certificateCache = new Hashtable<>();
```

Comment:
remove, this is not used anywhere

Reply:
Based this class off of SQLServerAASEnclaveProvider, and forgot to remove some stuff. Fixed.